tests/js/client/aql/aql-memory-limit.js

```diff
@@ -711,164 +719,252 @@ function ahuacatMemoryLimitSortedCollectTestSuite() {
       assertEqual(1, res.length);
       assertTrue(res[0] !== null);
     },
 
-    testMergeRecursiveArrayExceedLimit: function() {
+    testMergeRecursiveArrayExceedLimit: function () {
       const query = `RETURN MERGE_RECURSIVE((FOR i IN 1..2048 RETURN { [CONCAT('k', i)]: REPEAT('x', 4096), nested: { [CONCAT('nested', i)]: REPEAT('x', 4096) } }))`;
       try {
-        db._query(query, null, { memoryLimit: 32 * 1024 * 1024 });
+        db._query(query, null, {memoryLimit: 32 * 1024 * 1024});
         fail();
       } catch (err) {
         assertEqual(errors.ERROR_RESOURCE_LIMIT.code, err.errorNum);
       }
-    },
+    }
   };
 }
 
+function ahuacatMemoryLimitCollectMemoryLeakTestSuite () {
+  return {
+    testHitMemoryLimitAggregatorMin: function () {
+      const query = `
+       FOR d IN 1..200
+          LET big = REPEAT('a', 4096 + d)  
+          SORT d DESC
+          COLLECT g = d % 10
+          AGGREGATE m = MIN(big)
+          RETURN [g, LENGTH(m)]                     
+      `;
+      db._explain(query);
+      db._query(query);
+    },
+
+    testHitMemoryLimitAggregatorMin2: function () {
+      const query = `
+       FOR d IN 1..200
+          LET big = REPEAT('a', 4096 + d)  
+          LET n = d % 10 
+          SORT n ASC
+          COLLECT g = n
+          AGGREGATE m = MIN(big)
+          RETURN [g, LENGTH(m)]                     
+      `;
+      db._explain(query);
+      db._query(query);
+    },
+
+    testHitMemoryLimitAggregatorMax: function () {
+      const query = `
+       FOR d IN 1..200
+          LET big = REPEAT('a', 4096 + d)  
+          SORT d DESC
+          COLLECT g = d % 10
+          AGGREGATE m = MIN(big)
+          RETURN [g, LENGTH(m)]                     
+      `;
+      db._explain(query);
+      db._query(query);
+    },
+
+    testHitMemoryLimitAggregatorMax2: function () {
+      const query = `
+       FOR d IN 1..200
+          LET big = REPEAT('a', 4096 + d)  
+          LET n = d % 10 
+          SORT n ASC
+          COLLECT g = n
+          AGGREGATE m = MIN(big)
+          RETURN [g, LENGTH(m)]                     
+      `;
+      db._explain(query);
+      db._query(query);
+    },
+
+    testHitMemoryLimitAggregatorMinSorted: function () {
+      const query = `
+    	FOR d IN 1..200
+      LET n = d % 10
+      LET big = REPEAT('a', 4096 + d)
+      SORT n ASC
+      COLLECT g = n
+      AGGREGATE m = MIN(big)
+      OPTIONS { method: "sorted" }
+      RETURN [g, LENGTH(m)]
+  `;
+      db._explain(query);
+      db._query(query);
+    },
+
+    testHitMemoryLimitAggregatorMaxSorted: function () {
+      const query = `
+    FOR d IN 1..200
+      LET n = d % 10
+      LET big = REPEAT('a', 4096 + d)
+      SORT n ASC
+      COLLECT g = n
+      AGGREGATE m = MAX(big)
+      OPTIONS { method: "sorted" }
+      RETURN [g, LENGTH(m)]
+  `;
+      db._explain(query);
+      db._query(query, {});
```

Comment:
There are still several `explain` that should be removed. Also, the tests don't have any assertions so it is not entirely clear what they are supposed to test. If we just expect the tests to run through successfully without issues, perhaps add an according comment to the test class.